### PR TITLE
Feature/prototype protobuffer data distribution

### PR DESCRIPTION
**Pull request by scfitze**
* **Tickets addressed:** 55 and 242
* **Review:** By commit
* **Merge strategy:** Merge (no squash)

## Description
This feature should add the ability to distribute data from the protobuffer in order to create the scene (spacecraft, camera, and celestial bodies) with correct positions and orientations.  

The first three commits are cleanups for the level. The fourth commit was added because the compiler was complaining about the order of include statements in the protobuf reader class. After that, the kinematics library was added to include useful functions for later. The next few commits included adding classes and blueprints for objects that will be spawned into the scene. After that, the protobuf actor class was modified to spawn in the objects and make sure the proper conversions were happening for object's positions and rotations. The last commit includes adding a skybox with a debug material to help identify objects in the scene.

## Verification
Validated by viewing that the camera, spacecraft, and all the celestial bodies are in the scene with their positions and rotations being accurate to the binary data.

## Documentation
No Documentation needed.

## Future work
Might be a good idea to add some unit tests to check that all the conversions between MRP, Quaternion, and DCM are accurate. There is also a significant amount of artifacts with the lighting on asteroids as well as some smearing of objects as positions are updated. Might be worth looking into a smooth way to interpolate between positions and figure out best practices for accurate directional light.

### CODE REVIEW

File: Source/cielim/Public/CelestialBody.h
Comment (patkenneally) [inline Source/cielim/Public/CelestialBody.h]: :small_red_triangle_down: Remove commented out code.

File: Source/cielim/Public/CelestialBody.h
Comment (patkenneally) [inline Source/cielim/Public/CelestialBody.h]: :small_red_triangle_down: Remove this text.

File: Source/cielim/Private/CelestialBody.cpp
Comment (patkenneally) [inline Source/cielim/Private/CelestialBody.cpp]: :small_red_triangle_down: Remove

File: Source/cielim/Private/CelestialBody.cpp
Comment (patkenneally) [inline Source/cielim/Private/CelestialBody.cpp]: :large_orange_diamond: Remove extra new line.

File: Source/cielim/Private/CelestialBody.cpp
Comment (patkenneally) [inline Source/cielim/Private/CelestialBody.cpp]: :large_orange_diamond: Extra new line

File: Source/cielim/Private/Spacecraft.cpp
Comment (patkenneally) [inline Source/cielim/Private/Spacecraft.cpp]: :small_red_triangle_down: Remove text

File: Source/cielim/Private/Spacecraft.cpp
Comment (patkenneally) [inline Source/cielim/Private/Spacecraft.cpp]: :small_red_triangle_down: I'll not continue writing it after this instance. You can remove the redundant new line all throughout.

File: Source/cielim/Private/ProtobufActor.cpp
Comment (patkenneally) [inline Source/cielim/Private/ProtobufActor.cpp]: :small_red_triangle_down: We can invert this logic to make the function read simpler. Check if not blue print, emit error and return, no else condition, just continue on with the function.

File: Source/cielim/Private/ProtobufActor.cpp
Comment (patkenneally) [inline Source/cielim/Private/ProtobufActor.cpp]: :small_red_triangle_down: Remove commented out code.

File: Source/cielim/Private/ProtobufActor.cpp
Comment (patkenneally) [inline Source/cielim/Private/ProtobufActor.cpp]: :small_red_triangle_down: You can remove the comments in the function. They don't say anything the code isn't already directly saying.

File: Source/cielim/Private/ProtobufActor.cpp
Comment (patkenneally) [inline Source/cielim/Private/ProtobufActor.cpp]: :small_red_triangle_down: There is some odd indentation throughout the files in this PR. Maybe IDE settings?

**Review by patkenneally — CHANGES_REQUESTED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

I added some comments and submitted them so as to not hold you up. I do want to pull, build, and run the branch locally. I'll do that later today.

File: Source/cielim/Public/CelestialBody.h
Comment (scfitze) [reply Source/cielim/Public/CelestialBody.h]: Fixed

File: Source/cielim/Public/CelestialBody.h
Comment (scfitze) [reply Source/cielim/Public/CelestialBody.h]: Fixed

File: Source/cielim/Private/CelestialBody.cpp
Comment (scfitze) [reply Source/cielim/Private/CelestialBody.cpp]: Fixed

File: Source/cielim/Private/CelestialBody.cpp
Comment (scfitze) [reply Source/cielim/Private/CelestialBody.cpp]: Fixed

File: Source/cielim/Private/CelestialBody.cpp
Comment (scfitze) [reply Source/cielim/Private/CelestialBody.cpp]: Fixed

File: Source/cielim/Private/Spacecraft.cpp
Comment (scfitze) [reply Source/cielim/Private/Spacecraft.cpp]: Fixed

File: Source/cielim/Private/ProtobufActor.cpp
Comment (scfitze) [reply Source/cielim/Private/ProtobufActor.cpp]: Fixed and did the same logic for spawning celestial bodies

File: Source/cielim/Private/ProtobufActor.cpp
Comment (scfitze) [reply Source/cielim/Private/ProtobufActor.cpp]: Fixed

File: Source/cielim/Private/ProtobufActor.cpp
Comment (scfitze) [reply Source/cielim/Private/ProtobufActor.cpp]: Fixed

File: Source/cielim/Private/ProtobufActor.cpp
Comment (scfitze) [reply Source/cielim/Private/ProtobufActor.cpp]: This one was weird because I did not see the indentation on my editor, Ill check if this is still an issue

**Review by patkenneally — APPROVED** [review]

**Pull request merged**
